src/python/geoclaw/surge/storm.py

```diff
@@ -548,6 +549,115 @@ def read_ibtracs(self, path, storm_name, year):
             self.storm_radius = numpy.where(ds.usa_roci >=0,
                 units.convert(ds.usa_roci.values,'nmi','m'),-1)
 
+    def read_emanuel(self, path, storm_name, verbose=False):
+        r"""Read in Kerry Emanuel's storm file
+
+        This reads in the netcdf-formatted tracks from Kerry Emanuel.
+        Correspondence September 2018. Original storms have been converted
+        from matlab to netcdf format by mlimb@rhg.com and are at
+        /gcs/../coastal/storm_tracks/emmanuel_projections_2018-09-12/netcdf_added_velocity/.
+
+        :Input:
+        - *path* (string) Path to the file to be read.
+        - *storm_name* (string) storm name containing information about
+        model, scenario, time period, and storm index within the file.
+            ex. ccsm4_rcp85_2007_2025_0, hadgem5_rcp45_2035_2045_10
+            model_scenario_period is equivalent to filename.
+
+        Assume these attributes exist in Emanuel dataset:
+           datetime
+           longstore
+           latstore
+           rmstrore
+           pstore (central pressure)
+           v_total_max_ms
+           bas (attribute)
+
+        :Raises:
+        - ImportError if xarray not found
+        - KeyError if provided storm name is not found in
+        given path (assumes path is correct)
+        """
+        # try/except copied from read_ibtracs
+        # imports that you don't need for other read functions
+        try:
+            import xarray as xr
+        except ImportError as e:
+            print("Emanuel tracks currently require xarray "
+                  "to work.")
+            raise e
+
+        def _convert_to_python_datetime(np_date):
+            '''Convert numpy.datetime64 to python datetime.datetime'''
+            (year, month, day_hour) = str(np_date.values).split('-')
+            day = day_hour[0:2]
+            hour = day_hour[3:5]
+            return datetime.datetime(int(year), int(month), int(day), int(hour))
```

Comment:
why not just use Ian's method from above? converting a whole array to string and then back to int is pretty slow.

Reply:
should be something like
```python
self.t = list(map(
    lambda x: datetime.datetime(*x),
    zip(t.dt.year,t.dt.month,t.dt.day,t.dt.hour,t.dt.minute,t.dt.second)))
```

Reply:
Where'd we land in this. Should we change the numpy datetime to datetime.datetime approach?

Reply:
it seems keeping the current function actually seems faster than creating a new DataArray

I tried two other methods (based on implementation in read_ibtracs):
1) ```
dts = xr.DataArray.from_series(to_datetime(
                storm.datetime.astype(str).to_series()))
            
self.t = [datetime.datetime(
    t.dt.year, t.dt.month, t.dt.day,
    t.dt.minute, t.dt.second) for t in dts]
```

2) ```
   t = xr.DataArray.from_series(to_datetime(
                storm.datetime.astype(str).to_series()))
            
  self.t = list(map(
        lambda x: datetime.datetime(*x),
        zip(t.dt.year,t.dt.month,t.dt.day,t.dt.hour,t.dt.minute,t.dt.second)))
```
based on profiling, both were slower than the current method.

miroc = Storm()
%timeit miroc.read_emanuel(miroc_file, 'miroc5_rcp45_2055_2065_99')```
current method: 58.5 ms ± 745 µs
1) method: 199 ms ± 6 ms
2) method: 134 ms ± 1.13 ms
this storm has 73 time steps

```%timeit ccsm.read_emanuel(ccsm_file, 'ccsm4_rcp85_2035_2045_99')```
current method: 43.3 ms ± 506 µs
1) method: 79.2 ms ± 512 µs 
2) method: 63.2 ms ± 1.76 ms
this storm has 16 time steps

Reply:
@delgadom @bolliger32 so I think we can keep the current implementation?

Reply:
yeah all of these are many orders of magnitude smaller than the actual geoclaw run. Plenty fast! Nice work!